mailbox.c

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright (c) 2012, Broadcom Europe Ltd.
```

Comment:
I took it from richardghirst proposal for servoblaster and modified it some (added the board_model/firmware_revision functions, added debug output,  ... )

But it can be  found in other places, e.g. here from the raspberrypi repositories: https://github.com/raspberrypi/userland/blob/master/host_applications/linux/apps/hello_pi/hello_fft/mailbox.c